modules/gapi/test/infer/gapi_infer_ov_tests.cpp

```diff
@@ -657,6 +658,49 @@ TEST_F(TestAgeGenderListOV, InferList2Generic_Image) {
     validate();
 }
 
+TEST(TestOV, Infer_ImageCorrectNHWCInputLayout) {
+    const std::string model_name   = "ModelNHWC";
+    const std::string model_path   = model_name + ".xml";
+    const std::string weights_path = model_name + ".bin";
+    const std::string device_id    = "CPU";
+    const int W                    = 128;
+    const int H                    = 64;
+    const int C                    = 3;
+    const int N                    = 1;
+
+    auto data1 = std::make_shared<ov::opset8::Parameter>(ov::element::u8, ov::Shape{N, H, W, C});
+    data1->output(0).set_names({"data1_t"});  // tensor names
+    ov::layout::set_layout(data1, ov::Layout("NHWC"));
+
+    auto sin = std::make_shared<ov::opset8::Sin>(data1);
+    sin->output(0).set_names({"sin_t"});  // tensor name
+
+    auto result = std::make_shared<ov::opset8::Result>(sin);
+    result->output(0).set_names({"result_t"});  // tensor name
+
+    auto f = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{data1}, "function_name");
+
+    ov::serialize(f, model_path, weights_path, ov::pass::Serialize::Version::IR_V11);
+
+    cv::Mat in_mat1(std::vector<int>{N, C, 32, 256}, CV_8U), gapi_mat;
+    cv::randu(in_mat1, 0, 100);
+
+    cv::GMat g_in1;
+    cv::GInferInputs inputs;
+    inputs["data1_t"] = g_in1;
+    auto outputs = cv::gapi::infer<cv::gapi::Generic>(model_name, inputs);
+    auto out = outputs.at("result_t");
+
+    cv::GComputation comp(cv::GIn(g_in1), cv::GOut(out));
+
+    auto pp = cv::gapi::ov::Params<cv::gapi::Generic>(model_name,
+                                                      model_path,
+                                                      weights_path,
+                                                      device_id);
+    EXPECT_NO_THROW(comp.apply(cv::gin(in_mat1), cv::gout(gapi_mat),
```

Comment:
Tbh I didn't get the purpose of this test, the model has `NHWC` layout but you pass `NCHW` mat, so it won't throw any exceptions because `ov` backend doesn't validate either layouts match or not.

The test should be something like that:
1. Model has `NHWC` layout
2. Provide 4D mat and specify only `tensor` layout (e.g cfgInputTensorLayout('NCHW'))
3. Check if there is `NHWC->NCHW` conversion is added to the model. (it won't be added without your changes, since model layout not specified therefore won't be propagated into `PrePostProcessor`)